### PR TITLE
QUIC: CID conformance

### PR DESCRIPTION
**Pull request by paulidale**
- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Review by t8m — CHANGES_REQUESTED** [review]

File: ssl/quic/quic_channel.c
Comment (paulidale) [reply ssl/quic/quic_channel.c:2194]: It's going to need reworking once we properly support connections.  I'd prefer to leave until then.

**Review by t8m — APPROVED** [review]

File: ssl/quic/quic_channel.c
Comment (paulidale) [reply ssl/quic/quic_channel.c]: I just removed the check -- we can't validate out of order packet seq_no increase by one IMO.
We also don't send new CID packets, so there is nothing needed for this.

**Comment by paulidale** [comment]
Ping @openssl/committers for second review please.

File: ssl/quic/quic_channel.c
Comment (t8m) [reply ssl/quic/quic_channel.c:2205]: Yes, it is, see the code above.

**Comment by paulidale** [comment]
Merged, thanks for the comments

**Pull request closed without merging**
